[MNOE-574] - Improve error management

x4d3 opened the pull request:
I thought we could improve the error management in MNOE by using a custom error catched on the controller level.

@BrunoChauvet  @alachaum  what do you think ?

This is waiting on #521 to be merged

It is also waiting on https://github.com/chingor13/json_api_client/pull/275

BrunoChauvet commented:
I like this way of handling errors, way more concise and we end-up with leaner controllers :+1:

x4d3 commented:
@ouranos  PR ready for review

x4d3 commented on core/lib/mno_enterprise/concerns/models/team.rb:
```diff
@@ -24,29 +24,6 @@ module ClassMethods
   #==================================================================
   # Instance methods
   #==================================================================
-  # Add a user to the team
```
Yes it is, they are actually not called in the current code.

x4d3 commented:
@ouranos  I've addressed your comment. Regarding the error management, it's indeed changing, but I don't think we were using the result array or hash properly on the frontend.

ouranos commented:
It's not used properly at the moment but @alexnoox was looking into it.
The previous format was structured which would make it easy to parse and format in the frontend.
This new one, not so much.

x4d3 commented:
@ouranos  I've addressed your concerns. The json returned will be the same and follow that format
```{"name":["is too short (minimum is 2 characters)"]}```

alexnoox commented:
Good work, it is much cleaner!

For the error handler in the frontends, we should do it in an interceptor checking if the response is an error. That would allow us to remove all the error management in the services. The errors will need to be re-thrower from the handler to catch them in the controllers if we need to display a toastr or errors on a form.

x4d3 commented:
@ouranos I've fixed the specs.
